.github/workflows/test.yml

```diff
@@ -35,6 +35,7 @@ jobs:
       matrix:
         type:
           [
+            shellcheck,
             lint,
```

Comment:
I have the impression that for many linting jobs the setup on GitHub CI will take more time than the execution of the linter.

Mind checking if this is true for shellcheck? If so, I suggest modifying `make lint` to also execute this one.